[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
[NbConvertApp] Converting notebook mpi.ipynb to notebook
Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 154, in wrapped
Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 154, in wrapped
Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 154, in wrapped
Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 154, in wrapped
Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 154, in wrapped
    asyncio.get_running_loop()
    asyncio.get_running_loop()
    asyncio.get_running_loop()
    asyncio.get_running_loop()
    asyncio.get_running_loop()
RuntimeError: no running event loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
RuntimeError: no running event loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
RuntimeError: no running event loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/bin/jupyter-nbconvert", line 7, in <module>
RuntimeError: no running event loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/bin/jupyter-nbconvert", line 7, in <module>
RuntimeError: no running event loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/bin/jupyter-nbconvert", line 7, in <module>
  File "/home/fcastro/.conda/envs/desc-ray/bin/jupyter-nbconvert", line 7, in <module>
  File "/home/fcastro/.conda/envs/desc-ray/bin/jupyter-nbconvert", line 7, in <module>
    sys.exit(main())
    sys.exit(main())
             ^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/application.py", line 284, in launch_instance
    sys.exit(main())
    sys.exit(main())
    sys.exit(main())
             ^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/application.py", line 284, in launch_instance
             ^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/application.py", line 284, in launch_instance
             ^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/application.py", line 284, in launch_instance
             ^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/application.py", line 284, in launch_instance
    super().launch_instance(argv=argv, **kwargs)
    super().launch_instance(argv=argv, **kwargs)
    super().launch_instance(argv=argv, **kwargs)
    super().launch_instance(argv=argv, **kwargs)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/traitlets/config/application.py", line 1075, in launch_instance
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/traitlets/config/application.py", line 1075, in launch_instance
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/traitlets/config/application.py", line 1075, in launch_instance
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/traitlets/config/application.py", line 1075, in launch_instance
    super().launch_instance(argv=argv, **kwargs)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/traitlets/config/application.py", line 1075, in launch_instance
    app.start()
    app.start()
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 420, in start
    app.start()
    app.start()
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 420, in start
    app.start()
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 420, in start
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 420, in start
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 420, in start
    self.convert_notebooks()
    self.convert_notebooks()
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 597, in convert_notebooks
    self.convert_notebooks()
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 597, in convert_notebooks
    self.convert_notebooks()
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 597, in convert_notebooks
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 597, in convert_notebooks
    self.convert_notebooks()
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 597, in convert_notebooks
    self.convert_single_notebook(notebook_filename)
    self.convert_single_notebook(notebook_filename)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 563, in convert_single_notebook
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 563, in convert_single_notebook
    self.convert_single_notebook(notebook_filename)
    self.convert_single_notebook(notebook_filename)
    self.convert_single_notebook(notebook_filename)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 563, in convert_single_notebook
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 563, in convert_single_notebook
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 563, in convert_single_notebook
    output, resources = self.export_single_notebook(
    output, resources = self.export_single_notebook(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 487, in export_single_notebook
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 487, in export_single_notebook
    output, resources = self.export_single_notebook(
    output, resources = self.export_single_notebook(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 487, in export_single_notebook
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 487, in export_single_notebook
    output, resources = self.export_single_notebook(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 487, in export_single_notebook
    output, resources = self.exporter.from_filename(
    output, resources = self.exporter.from_filename(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 201, in from_filename
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 201, in from_filename
    output, resources = self.exporter.from_filename(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    output, resources = self.exporter.from_filename(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 201, in from_filename
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 201, in from_filename
    output, resources = self.exporter.from_filename(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 201, in from_filename
    return self.from_file(f, resources=resources, **kw)
    return self.from_file(f, resources=resources, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 220, in from_file
    return self.from_file(f, resources=resources, **kw)
    return self.from_file(f, resources=resources, **kw)
    return self.from_file(f, resources=resources, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 220, in from_file
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 220, in from_file
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 220, in from_file
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 220, in from_file
    return self.from_notebook_node(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/notebook.py", line 36, in from_notebook_node
    return self.from_notebook_node(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/notebook.py", line 36, in from_notebook_node
    return self.from_notebook_node(
    return self.from_notebook_node(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/notebook.py", line 36, in from_notebook_node
    return self.from_notebook_node(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/notebook.py", line 36, in from_notebook_node
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/notebook.py", line 36, in from_notebook_node
    nb_copy, resources = super().from_notebook_node(nb, resources, **kw)
    nb_copy, resources = super().from_notebook_node(nb, resources, **kw)
    nb_copy, resources = super().from_notebook_node(nb, resources, **kw)
    nb_copy, resources = super().from_notebook_node(nb, resources, **kw)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 154, in from_notebook_node
    nb_copy, resources = super().from_notebook_node(nb, resources, **kw)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 154, in from_notebook_node
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 154, in from_notebook_node
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 154, in from_notebook_node
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 154, in from_notebook_node
    nb_copy, resources = self._preprocess(nb_copy, resources)
    nb_copy, resources = self._preprocess(nb_copy, resources)
    nb_copy, resources = self._preprocess(nb_copy, resources)
    nb_copy, resources = self._preprocess(nb_copy, resources)
    nb_copy, resources = self._preprocess(nb_copy, resources)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 353, in _preprocess
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 353, in _preprocess
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 353, in _preprocess
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 353, in _preprocess
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 353, in _preprocess
    nbc, resc = preprocessor(nbc, resc)
    nbc, resc = preprocessor(nbc, resc)
    nbc, resc = preprocessor(nbc, resc)
    nbc, resc = preprocessor(nbc, resc)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/base.py", line 48, in __call__
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/base.py", line 48, in __call__
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/base.py", line 48, in __call__
    nbc, resc = preprocessor(nbc, resc)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/base.py", line 48, in __call__
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/base.py", line 48, in __call__
    return self.preprocess(nb, resources)
    return self.preprocess(nb, resources)
    return self.preprocess(nb, resources)
    return self.preprocess(nb, resources)
    return self.preprocess(nb, resources)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 103, in preprocess
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 103, in preprocess
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 103, in preprocess
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 103, in preprocess
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 103, in preprocess
Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 154, in wrapped
    asyncio.get_running_loop()
RuntimeError: no running event loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/bin/jupyter-nbconvert", line 7, in <module>
    self.preprocess_cell(cell, resources, index)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 124, in preprocess_cell
    self.preprocess_cell(cell, resources, index)
    sys.exit(main())
    self.preprocess_cell(cell, resources, index)
    self.preprocess_cell(cell, resources, index)
    self.preprocess_cell(cell, resources, index)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 124, in preprocess_cell
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 124, in preprocess_cell
             ^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/application.py", line 284, in launch_instance
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 124, in preprocess_cell
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 124, in preprocess_cell
    cell = self.execute_cell(cell, index, store_history=True)
    cell = self.execute_cell(cell, index, store_history=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 158, in wrapped
    cell = self.execute_cell(cell, index, store_history=True)
    cell = self.execute_cell(cell, index, store_history=True)
    cell = self.execute_cell(cell, index, store_history=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 158, in wrapped
    super().launch_instance(argv=argv, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 158, in wrapped
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 158, in wrapped
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 158, in wrapped
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/traitlets/config/application.py", line 1075, in launch_instance
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/asyncio/base_events.py", line 691, in run_until_complete
    return loop.run_until_complete(inner)
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/asyncio/base_events.py", line 691, in run_until_complete
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/asyncio/base_events.py", line 691, in run_until_complete
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/asyncio/base_events.py", line 691, in run_until_complete
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/asyncio/base_events.py", line 691, in run_until_complete
    app.start()
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 420, in start
    self.convert_notebooks()
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 597, in convert_notebooks
    self.convert_single_notebook(notebook_filename)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 563, in convert_single_notebook
    output, resources = self.export_single_notebook(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 487, in export_single_notebook
    output, resources = self.exporter.from_filename(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 201, in from_filename
    return self.from_file(f, resources=resources, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 220, in from_file
    return self.from_notebook_node(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/notebook.py", line 36, in from_notebook_node
    nb_copy, resources = super().from_notebook_node(nb, resources, **kw)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 154, in from_notebook_node
    nb_copy, resources = self._preprocess(nb_copy, resources)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 353, in _preprocess
    nbc, resc = preprocessor(nbc, resc)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/base.py", line 48, in __call__
    return self.preprocess(nb, resources)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 103, in preprocess
    self.preprocess_cell(cell, resources, index)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 124, in preprocess_cell
    cell = self.execute_cell(cell, index, store_history=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 158, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/asyncio/base_events.py", line 691, in run_until_complete
Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 154, in wrapped
    asyncio.get_running_loop()
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
RuntimeError: no running event loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/bin/jupyter-nbconvert", line 7, in <module>
    return future.result()
    return future.result()
    return future.result()
           ^^^^^^^^^^^^^^^
           ^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    return future.result()
           ^^^^^^^^^^^^^^^
    return future.result()
           ^^^^^^^^^^^^^^^
           ^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    sys.exit(main())
             ^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/application.py", line 284, in launch_instance
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    super().launch_instance(argv=argv, **kwargs)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/traitlets/config/application.py", line 1075, in launch_instance
    app.start()
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 420, in start
    self.convert_notebooks()
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 597, in convert_notebooks
    self.convert_single_notebook(notebook_filename)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 563, in convert_single_notebook
    output, resources = self.export_single_notebook(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 487, in export_single_notebook
    output, resources = self.exporter.from_filename(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 201, in from_filename
    return self.from_file(f, resources=resources, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 220, in from_file
    return self.from_notebook_node(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/notebook.py", line 36, in from_notebook_node
    nb_copy, resources = super().from_notebook_node(nb, resources, **kw)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 154, in from_notebook_node
    nb_copy, resources = self._preprocess(nb_copy, resources)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 353, in _preprocess
    nbc, resc = preprocessor(nbc, resc)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/base.py", line 48, in __call__
    return self.preprocess(nb, resources)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 103, in preprocess
    self.preprocess_cell(cell, resources, index)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 124, in preprocess_cell
    cell = self.execute_cell(cell, index, store_history=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 158, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/asyncio/base_events.py", line 691, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    await self._check_raise_for_error(cell, cell_index, exec_reply)
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    await self._check_raise_for_error(cell, cell_index, exec_reply)
    await self._check_raise_for_error(cell, cell_index, exec_reply)
    await self._check_raise_for_error(cell, cell_index, exec_reply)
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
sgrid2 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*1)).T
                    )
sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*2)).T
                    )

sdata2 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid2)
sdata2 = iso_coords_interp(tdata, sdata2, sgrid2)

sdata3 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid3)
sdata3 = iso_coords_interp(tdata, sdata3, sgrid3)
------------------


[31m---------------------------------------------------------------------------[39m
[31mNameError[39m                                 Traceback (most recent call last)
[36mCell[39m[36m [39m[32mIn[5][39m[32m, line 1[39m
[32m----> [39m[32m1[39m sgrid2 = Grid(nodes = jnp.vstack(([43msgrid[49m.nodes[:,[32m0[39m],
[32m      2[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      3[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m1[39m)).T
[32m      4[39m                     )
[32m      5[39m sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,[32m0[39m],
[32m      6[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      7[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m2[39m)).T
[32m      8[39m                     )
[32m     10[39m sdata2 = surf_winding.compute([[33m"[39m[33mtheta[39m[33m"[39m,[33m"[39m[33mzeta[39m[33m"[39m, [33m"[39m[33me^theta_s[39m[33m"[39m,[33m"[39m[33me^zeta_s[39m[33m"[39m,[33m'[39m[33mx[39m[33m'[39m], grid = sgrid2)

[31mNameError[39m: name 'sgrid' is not defined

    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
sgrid2 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*1)).T
                    )
sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*2)).T
                    )

sdata2 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid2)
sdata2 = iso_coords_interp(tdata, sdata2, sgrid2)

sdata3 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid3)
sdata3 = iso_coords_interp(tdata, sdata3, sgrid3)
------------------


[31m---------------------------------------------------------------------------[39m
[31mNameError[39m                                 Traceback (most recent call last)
[36mCell[39m[36m [39m[32mIn[5][39m[32m, line 1[39m
[32m----> [39m[32m1[39m sgrid2 = Grid(nodes = jnp.vstack(([43msgrid[49m.nodes[:,[32m0[39m],
[32m      2[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      3[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m1[39m)).T
[32m      4[39m                     )
[32m      5[39m sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,[32m0[39m],
[32m      6[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      7[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m2[39m)).T
[32m      8[39m                     )
[32m     10[39m sdata2 = surf_winding.compute([[33m"[39m[33mtheta[39m[33m"[39m,[33m"[39m[33mzeta[39m[33m"[39m, [33m"[39m[33me^theta_s[39m[33m"[39m,[33m"[39m[33me^zeta_s[39m[33m"[39m,[33m'[39m[33mx[39m[33m'[39m], grid = sgrid2)

[31mNameError[39m: name 'sgrid' is not defined

nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
sgrid2 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*1)).T
                    )
sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*2)).T
                    )

sdata2 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid2)
sdata2 = iso_coords_interp(tdata, sdata2, sgrid2)

sdata3 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid3)
sdata3 = iso_coords_interp(tdata, sdata3, sgrid3)
------------------


[31m---------------------------------------------------------------------------[39m
[31mNameError[39m                                 Traceback (most recent call last)
[36mCell[39m[36m [39m[32mIn[5][39m[32m, line 1[39m
[32m----> [39m[32m1[39m sgrid2 = Grid(nodes = jnp.vstack(([43msgrid[49m.nodes[:,[32m0[39m],
[32m      2[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      3[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m1[39m)).T
[32m      4[39m                     )
[32m      5[39m sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,[32m0[39m],
[32m      6[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      7[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m2[39m)).T
[32m      8[39m                     )
[32m     10[39m sdata2 = surf_winding.compute([[33m"[39m[33mtheta[39m[33m"[39m,[33m"[39m[33mzeta[39m[33m"[39m, [33m"[39m[33me^theta_s[39m[33m"[39m,[33m"[39m[33me^zeta_s[39m[33m"[39m,[33m'[39m[33mx[39m[33m'[39m], grid = sgrid2)

[31mNameError[39m: name 'sgrid' is not defined

    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
sgrid2 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*1)).T
                    )
sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*2)).T
                    )

sdata2 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid2)
sdata2 = iso_coords_interp(tdata, sdata2, sgrid2)

sdata3 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid3)
sdata3 = iso_coords_interp(tdata, sdata3, sgrid3)
------------------


[31m---------------------------------------------------------------------------[39m
[31mNameError[39m                                 Traceback (most recent call last)
[36mCell[39m[36m [39m[32mIn[5][39m[32m, line 1[39m
[32m----> [39m[32m1[39m sgrid2 = Grid(nodes = jnp.vstack(([43msgrid[49m.nodes[:,[32m0[39m],
[32m      2[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      3[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m1[39m)).T
[32m      4[39m                     )
[32m      5[39m sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,[32m0[39m],
[32m      6[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      7[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m2[39m)).T
[32m      8[39m                     )
[32m     10[39m sdata2 = surf_winding.compute([[33m"[39m[33mtheta[39m[33m"[39m,[33m"[39m[33mzeta[39m[33m"[39m, [33m"[39m[33me^theta_s[39m[33m"[39m,[33m"[39m[33me^zeta_s[39m[33m"[39m,[33m'[39m[33mx[39m[33m'[39m], grid = sgrid2)

[31mNameError[39m: name 'sgrid' is not defined

    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
sgrid2 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*1)).T
                    )
sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*2)).T
                    )

sdata2 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid2)
sdata2 = iso_coords_interp(tdata, sdata2, sgrid2)

sdata3 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid3)
sdata3 = iso_coords_interp(tdata, sdata3, sgrid3)
------------------


[31m---------------------------------------------------------------------------[39m
[31mNameError[39m                                 Traceback (most recent call last)
[36mCell[39m[36m [39m[32mIn[5][39m[32m, line 1[39m
[32m----> [39m[32m1[39m sgrid2 = Grid(nodes = jnp.vstack(([43msgrid[49m.nodes[:,[32m0[39m],
[32m      2[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      3[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m1[39m)).T
[32m      4[39m                     )
[32m      5[39m sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,[32m0[39m],
[32m      6[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      7[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m2[39m)).T
[32m      8[39m                     )
[32m     10[39m sdata2 = surf_winding.compute([[33m"[39m[33mtheta[39m[33m"[39m,[33m"[39m[33mzeta[39m[33m"[39m, [33m"[39m[33me^theta_s[39m[33m"[39m,[33m"[39m[33me^zeta_s[39m[33m"[39m,[33m'[39m[33mx[39m[33m'[39m], grid = sgrid2)

[31mNameError[39m: name 'sgrid' is not defined

    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
sgrid2 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*1)).T
                    )
sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*2)).T
                    )

sdata2 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid2)
sdata2 = iso_coords_interp(tdata, sdata2, sgrid2)

sdata3 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid3)
sdata3 = iso_coords_interp(tdata, sdata3, sgrid3)
------------------


[31m---------------------------------------------------------------------------[39m
[31mNameError[39m                                 Traceback (most recent call last)
[36mCell[39m[36m [39m[32mIn[5][39m[32m, line 1[39m
[32m----> [39m[32m1[39m sgrid2 = Grid(nodes = jnp.vstack(([43msgrid[49m.nodes[:,[32m0[39m],
[32m      2[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      3[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m1[39m)).T
[32m      4[39m                     )
[32m      5[39m sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,[32m0[39m],
[32m      6[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      7[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m2[39m)).T
[32m      8[39m                     )
[32m     10[39m sdata2 = surf_winding.compute([[33m"[39m[33mtheta[39m[33m"[39m,[33m"[39m[33mzeta[39m[33m"[39m, [33m"[39m[33me^theta_s[39m[33m"[39m,[33m"[39m[33me^zeta_s[39m[33m"[39m,[33m'[39m[33mx[39m[33m'[39m], grid = sgrid2)

[31mNameError[39m: name 'sgrid' is not defined

    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
sgrid2 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*1)).T
                    )
sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*2)).T
                    )

sdata2 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid2)
sdata2 = iso_coords_interp(tdata, sdata2, sgrid2)

sdata3 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid3)
sdata3 = iso_coords_interp(tdata, sdata3, sgrid3)
------------------


[31m---------------------------------------------------------------------------[39m
[31mNameError[39m                                 Traceback (most recent call last)
[36mCell[39m[36m [39m[32mIn[5][39m[32m, line 1[39m
[32m----> [39m[32m1[39m sgrid2 = Grid(nodes = jnp.vstack(([43msgrid[49m.nodes[:,[32m0[39m],
[32m      2[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      3[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m1[39m)).T
[32m      4[39m                     )
[32m      5[39m sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,[32m0[39m],
[32m      6[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      7[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m2[39m)).T
[32m      8[39m                     )
[32m     10[39m sdata2 = surf_winding.compute([[33m"[39m[33mtheta[39m[33m"[39m,[33m"[39m[33mzeta[39m[33m"[39m, [33m"[39m[33me^theta_s[39m[33m"[39m,[33m"[39m[33me^zeta_s[39m[33m"[39m,[33m'[39m[33mx[39m[33m'[39m], grid = sgrid2)

[31mNameError[39m: name 'sgrid' is not defined

Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 154, in wrapped
    asyncio.get_running_loop()
RuntimeError: no running event loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fcastro/.conda/envs/desc-ray/bin/jupyter-nbconvert", line 7, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/application.py", line 284, in launch_instance
    super().launch_instance(argv=argv, **kwargs)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/traitlets/config/application.py", line 1075, in launch_instance
    app.start()
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 420, in start
    self.convert_notebooks()
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 597, in convert_notebooks
    self.convert_single_notebook(notebook_filename)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 563, in convert_single_notebook
    output, resources = self.export_single_notebook(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/nbconvertapp.py", line 487, in export_single_notebook
    output, resources = self.exporter.from_filename(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 201, in from_filename
    return self.from_file(f, resources=resources, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 220, in from_file
    return self.from_notebook_node(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/notebook.py", line 36, in from_notebook_node
    nb_copy, resources = super().from_notebook_node(nb, resources, **kw)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 154, in from_notebook_node
    nb_copy, resources = self._preprocess(nb_copy, resources)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 353, in _preprocess
    nbc, resc = preprocessor(nbc, resc)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/base.py", line 48, in __call__
    return self.preprocess(nb, resources)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 103, in preprocess
    self.preprocess_cell(cell, resources, index)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 124, in preprocess_cell
    cell = self.execute_cell(cell, index, store_history=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 158, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/asyncio/base_events.py", line 691, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/fcastro/.conda/envs/desc-ray/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
sgrid2 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*1)).T
                    )
sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,0],
                                        sgrid.nodes[:,1],
                                        sgrid.nodes[:,2] + (2*np.pi/surf_winding.NFP)*2)).T
                    )

sdata2 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid2)
sdata2 = iso_coords_interp(tdata, sdata2, sgrid2)

sdata3 = surf_winding.compute(["theta","zeta", "e^theta_s","e^zeta_s",'x'], grid = sgrid3)
sdata3 = iso_coords_interp(tdata, sdata3, sgrid3)
------------------


[31m---------------------------------------------------------------------------[39m
[31mNameError[39m                                 Traceback (most recent call last)
[36mCell[39m[36m [39m[32mIn[5][39m[32m, line 1[39m
[32m----> [39m[32m1[39m sgrid2 = Grid(nodes = jnp.vstack(([43msgrid[49m.nodes[:,[32m0[39m],
[32m      2[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      3[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m1[39m)).T
[32m      4[39m                     )
[32m      5[39m sgrid3 = Grid(nodes = jnp.vstack((sgrid.nodes[:,[32m0[39m],
[32m      6[39m                                         sgrid.nodes[:,[32m1[39m],
[32m      7[39m                                         sgrid.nodes[:,[32m2[39m] + ([32m2[39m*np.pi/surf_winding.NFP)*[32m2[39m)).T
[32m      8[39m                     )
[32m     10[39m sdata2 = surf_winding.compute([[33m"[39m[33mtheta[39m[33m"[39m,[33m"[39m[33mzeta[39m[33m"[39m, [33m"[39m[33me^theta_s[39m[33m"[39m,[33m"[39m[33me^zeta_s[39m[33m"[39m,[33m'[39m[33mx[39m[33m'[39m], grid = sgrid2)

[31mNameError[39m: name 'sgrid' is not defined

srun: error: della-r3c4n2: tasks 0-7: Exited with exit code 1
srun: Terminating StepId=289309.0
srun: error: della-r3c4n3: tasks 8-15: Terminated
srun: Force Terminated StepId=289309.0
